shoop/admin/modules/orders/json_order_creator.py

```diff
@@ -121,27 +136,93 @@ def _process_lines(self, source, state):
             except Exception as exc:  # pragma: no cover
                 self.add_error(exc)
 
-    def _initialize_source_from_state(self, state, creator):
-        customer_data = state.pop("customer", None) or {}
-        shop_data = state.pop("shop", None) or {}
-        methods_data = state.pop("methods", None) or {}
-        customer = self.safe_get_first(Contact, pk=customer_data.get("id"))
-        shop = self.safe_get_first(Shop, pk=shop_data.pop("id", None))
+    def _create_contact_from_address(self, billing_address, is_company):
+        name = billing_address.get("name", None)
+        if not name:
+            self.add_error(
+                ValidationError(_("Name is required for customer"), code="no_name")
+            )
 
-        if not customer:
-            self.add_error(ValidationError(_("Please choose a valid customer."), code="no_customer"))
+        phone = billing_address.get("phone", "")
+        email = billing_address.get("email", "")
+        fields = {"name": name, "phone": phone, "email": email}
+        if is_company:
+            tax_number = billing_address.pop("tax_number", None)
+            if not tax_number:
+                self.add_error(
+                    ValidationError(_("Tax number is not set for company."), code="no_tax_number")
+                )
+                return
+
+            fields.update({"tax_number": tax_number})
+            customer = CompanyContact.objects.create(**fields)
+        else:
+            customer = PersonContact.objects.create(**fields)
+        return customer
 
+    def _create_address(self, address):
+        address_fields = [
+            "name", "tax_number", "phone", "email", "street", "street2", "postal_code", "city", "region", "country"
+        ]
+        fields = {}
+        for field in address_fields:
+            fields[field] = address.get(field, "")
+
+        address = Address.objects.filter(**fields).first()
```

Comment:
I think yes. I don't think it is ideal to have same address with same fields multiple times in db. Also address is immutable. Also this "get or create" is reached already in confirmation phase so .create(**fields) is not enough since we don't want to create same address twice in order creation. This "get or create" is possibly done multiple times per one order creation.